core/src/test/scala/org/apache/spark/scheduler/ReplayListenerSuite.scala

```diff
@@ -218,8 +219,17 @@ class ReplayListenerSuite extends SparkFunSuite with BeforeAndAfter with LocalSp
 
     // Verify the same events are replayed in the same order
     assert(sc.eventLogger.isDefined)
-    val originalEvents = sc.eventLogger.get.loggedEvents
-    val replayedEvents = eventMonster.loggedEvents
+    implicit val format = DefaultFormats
+    def exceptCase(a: JValue) = ( a \ "Event").extract[String] match {
+      // If we are logging stage executor metrics, there is a bulk call to logEvent with
+      // SparkListenerStageExecutorMetrics events via a Map.foreach. The Map.foreach bulk
+      // operation may not log the events with the same order. So here we should not compare
+      // SparkListenerStageExecutorMetrics here.
```

Comment:
This comment is based on my earlier analysis in the JIRA, which is not quite correct.